src/NCronJob/Execution/JobExecutionProgressMonitor.cs

```diff
@@ -0,0 +1,150 @@
+using System.Collections.Immutable;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace NCronJob;
+
+#pragma warning disable CA1707 // Identifiers should not contain underscores
+#pragma warning disable S101 // Types should be named in PascalCase
+public record ExecutionProgress_Unstable
+#pragma warning restore S101 // Types should be named in PascalCase
+#pragma warning restore CA1707 // Identifiers should not contain underscores
+{
+    internal ExecutionProgress_Unstable(JobRun run)
```

Comment:
``xxx_unstable` is a standard pattern in the nodejs ecosystem when you want to release early a work in progress (or an experimental feature) but want to be free to introduce any breaking change outside of a major major version release.

Basically, it's a "Use at your own risk, may change or even disappear without prior notice" warning label.

Reply:
To be fair, the `ExperimentalAttribute` would be enough for me. That signals already what is going on.

Reply:
Fixed